src/responsibleai/rai_analyse/_score_card/_rai_insight_data.py

```diff
@@ -283,6 +293,11 @@ def __init__(self, raiinsightdata, config):
             self.pos_label = str(self.data._classes[0])
             self.classes = [self.other_class, self.pos_label]
 
+    def validate_valid_metric_for_task_type(self):
+        for m in self.metrics:
+            if m in metric_task_map and metric_task_map[m].lower() != self.tasktype:
```

Comment:
is this `.lower()` needed here for `metric_task_map[m]`? If yes, do we want to add `.lower()` for `self.tasktype` as well?

Reply:
self.tasktype is assigned either "regression" or "classification" in the logic of data generation class constructor.